addons/hw_drivers/main.py

```diff
@@ -29,10 +29,16 @@
 
 drivers = []
 interfaces = {}
+interfaces_instance = set()
+"""Contain all instances of interfaces"""
 iot_devices = {}
 
 
 class Manager(Thread):
+    def __init__(self):
+        super().__init__()
+        self.previous_iot_devices = []
+
```

Comment:
why not more concisely define it in the class instead of redefining `__init__` 

```suggestion
class Manager(Thread):
    previous_iot_devices = []

```